### PR TITLE
Feature/setup get user query

### PR DESCRIPTION
**Pull request by MTCawthray**
# Thank you for pushing your code!
## What (if any) features are you implementing?
- This branch sets up our ```GET_USER``` query to allow us to fetch our user data from the back end. We imported the Apollo client into the project and wrapped our ```APP Component``` in ```ApolloProvider```
- Once fetched, we set our state of user to the fetched user data

## What (if anything) did you refactor?
- We originally tried to set up our Apollo client within the App component, but decided to move it up the file tree structure to index.js

## Were there any issues that arose?
- We had a few issues figuring out how to dynamically fetch a user based on the user's ID. We eventually figured out how to work with the new syntax to accomplish this.

## Is there anything that you need from your teammates?
- Now that we have the user data fetching on page load, we can send it to the ```Profile component``` to build out our user profile view!

## Any other comments, questions, or concerns?
- Great job on working through the issues we had today. GO TEAM!

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor

## How Has This Been Tested?
- [x] open index.html
- [x] console.log()
- [x] dev tools
- [ ] Cypress

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/App/App.css [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 .App-header {
   background-color: #282c34;
-  min-height: 100vh;
+  /* min-height: 100vh; */
```
Comment (hayleyw7) [inline src/components/App/App.css:18]: Much easier to read while testing

File: src/components/App/App.js [hunk 2]
```diff
@@ -1,15 +1,37 @@
 import './App.css';
-
+import { useState, useEffect } from 'react';
```
Comment (hayleyw7) [inline src/components/App/App.js:2]: Hooks!

File: src/components/App/App.js [hunk 3]
```diff
@@ -1,15 +1,37 @@
...
+import Profile from '../Profile/Profile';
+import UserDisplay from '../UserDisplay/UserDisplay';
+import {useQuery} from '@apollo/client';
```
Comment (hayleyw7) [inline src/components/App/App.js:5]: Apollo!

File: src/components/App/App.js [hunk 4]
```diff
@@ -1,15 +1,37 @@
...
+import {GET_USER} from '../../graphQL/queries';
```
Comment (hayleyw7) [inline src/components/App/App.js:6]: GraphQL!

File: src/graphQL/queries.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import {gql} from '@apollo/client';
+
+export const GET_USER = gql `
```
Comment (nataliemcintyre2021) [inline src/graphQL/queries.js:3]: This looks great! Excited to think through how we will grab specific user after log in.

File: src/components/App/App.js [hunk 6]
```diff
@@ -1,15 +1,37 @@
...
 
 
 
 function App() {
+  
+  const [user, setUser] = useState({});
+  const [friends, setFriends] = useState([]);
+  const [allUsers, setAllUsers] = useState([]);
+  
+  const {error, loading, data} = useQuery(GET_USER, {
```
Comment (hayleyw7) [inline src/components/App/App.js:17]: Success on the FE & BE connection.

File: src/components/App/App.js [hunk 7]
```diff
@@ -1,15 +1,37 @@
...
+    variables: {id: "1"}
+  })
+  //create a findFriend function to trigger when we click on a friend card, then pass the return as prop to profile. 
+
+  useEffect(() => {
+      console.log(data);
```
Comment (hayleyw7) [inline src/components/App/App.js:23]: IT WORKS!

File: src/components/App/App.js [hunk 8]
```diff
@@ -1,15 +1,37 @@
...
-    <div className="App">
-      <header className="App-header">
-        <h1>EnneaLink</h1>
-      </header>
-    </div>
+      <div className="App">
+        <header className="App-header">
+          <h1>EnneaLink</h1>
+        </header>
+        <Profile profileView={user} />
```
Comment (hayleyw7) [inline src/components/App/App.js:32]: We will populate this next.

File: src/components/App/App.js [hunk 9]
```diff
@@ -1,15 +1,37 @@
...
+        <UserDisplay friends={friends} allUsers={allUsers} />
```
Comment (hayleyw7) [inline src/components/App/App.js:33]: We will populate this in future steps.

File: src/components/UserDisplay/UserDisplay.js [hunk 10]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+
+const UserDisplay = ({ friends, allUsers }) => {
+
+  return (
+    <p> I'm a user display!</p>
```
Comment (hayleyw7) [inline src/components/UserDisplay/UserDisplay.js:6]: This shows up on the DOM.

File: src/graphQL/queries.js [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+  query getUserStats($id: ID!) {
```
Comment (hayleyw7) [inline src/graphQL/queries.js:4]: It's dynamic and works.

File: src/index.js [hunk 12]
```diff
@@ -3,24 +3,43 @@ import ReactDOM from 'react-dom';
...
+  // fetchOptions: {
+  //   mode: 'no-cors'
+  // }
+})
 
 ReactDOM.render(
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>,
+  <ApolloProvider client={client}>
```
Comment (nataliemcintyre2021) [inline src/index.js:37]: We might want to try to move this back to App again, or another component to clear up index.js? Maybe in refactor!

File: src/index.js [hunk 13]
```diff
@@ -3,24 +3,43 @@ import ReactDOM from 'react-dom';
...
+import {onError} from '@apollo/client/link/error';
 
-import {
-  ApolloClient,
-  InMemoryCache,
-  ApolloProvider,
-  useQuery,
-  gql
-} from "@apollo/client";
+const errorLink = onError(({graphqlErrors, networkError}) => {
```
Comment (hayleyw7) [inline src/index.js:15]: We can try moving these back to App.js later.

File: src/components/App/App.js [hunk 14]
```diff
@@ -1,15 +1,37 @@
...
 
 
 function App() {
+  
+  const [user, setUser] = useState({});
+  const [friends, setFriends] = useState([]);
+  const [allUsers, setAllUsers] = useState([]);
+  
+  const {error, loading, data} = useQuery(GET_USER, {
+    variables: {id: "1"}
```
Comment (nataliemcintyre2021) [inline src/components/App/App.js:18]: Thinking through where we will eventually grab this from once user logged in...

**Pull request merged**